### PR TITLE
[cdc-base]remove redundant source metrics register

### PR DESCRIPTION
**Pull request by loserwang1024**
### What is the purpose of the change
As described in  [#2403](https://github.com/ververica/flink-cdc-connectors/issues/2403)

### Brief Change Log
* remove redundant source metrics register: such as sourceIdleTime, currentEmitEventTimeLag,numRecordsInErrors
* remove meaningless metrics register such as pendingRecords which is always 0 now.
* no longer support for Flink 1.13
* throw error in MongoDBRecordEmitter#processElement

### CODE REVIEW

**Comment by loserwang1024** [comment]
@PatrickRen , CC

**Comment by yuxiqian** [comment]
@loserwang1024 I pulled this PR and noticed that when running MongoDB E2e test, it always reports `currentEmitEventTimeLag` metric as `-1`. Could you please help checking this?

截屏2023-09-14 13 19 35

截屏2023-09-14 13 18 16

**Comment by loserwang1024** [comment]
@yuxiqian Thanks a lot, I will add to this pr

File: README.md [hunk 1]
```diff
@@ -29,12 +29,12 @@ This README is meant as a brief walkthrough on the core features of CDC Connecto
 
 We need several steps to setup a Flink cluster with the provided connector.
 
-1. Setup a Flink cluster with version 1.12+ and Java 8+ installed.
```
Comment (PatrickRen) [inline README.md:32]: What about creating an individual issue dropping Flink 1.13 support? Currently the commit is linked to #2403, whose description is about metrics. I think dropping support of a Flink version is relatively a big change and it deserves an individual issue to get tracked.

**Review by PatrickRen — COMMENTED** [review]
@loserwang1024 Thanks for the patch! LGTM overall. I left a comment. 

BTW what about cleaning `MySQLSourceReaderMetrics` as well?

File: README.md (hunk 1 above)
Comment (loserwang1024) [reply README.md]: This sounds good, and I will remove it and put it in an individual issue.

**Review by PatrickRen — APPROVED** [review]
@loserwang1024 Thanks for the update. LGTM

**Pull request merged**